Fixed custom_access_token_attributes related errors

JeremyC-za opened the pull request:
### Summary

These changes (relating to #1602) should fix the error reported in #1647 and may affect/resolve #1642. The cause of #1647 was that `lib/doorkeeper#configure` needed to be run before we attempt to access the `access_token_model` and `access_grant_model`. When it hasn't run, the models haven't been loaded by `lib/doorkeeper#setup_orm_adapter` yet. 

I've also updated the error message for when custom attributes aren't recognized by the grant/token models to give the user a better idea of how to resolve the issue.

JeremyC-za commented:
@nbulaj please review when you have time 🙂

nbulaj commented on lib/doorkeeper/config/validations.rb:
```diff
@@ -55,11 +55,17 @@ def validate_token_reuse_limit
+        # the access token and grant models before we can access them.
+        Doorkeeper.configuration
+
         custom_access_token_attributes.each do |attribute_name|
```
Yeah it makes more sense to me to use `Doorkeeper.config.custom_access_token_attributes`

Also I'm wondering if we need this validation at all :thinking: We can use `model.column_names` and extract only those which are present, so if attribute isn't present in the model - it will be skipped.

JeremyC-za commented:
@nbulaj thoughts on these changes I've made? I've replaced the ConfigError with a warning instead, and removed the unrecognized attributes. 

I also realised we can't call `Doorkeeper.config` in the validation - it tries to validate the config and causes an infinite loop. So to fix the error from #1647 I've instead reordered `Doorkeeper#configure` to run `setup` before the configuration is validated, although I'm not sure if that has other impacts I'm unaware of.

nbulaj reviewed (commented):
I'm not sure these changes around configuration block is necessary (looks like a hack).

I also think that we don't need to validate custom attributes at all - I think all the developers should use correct values and test their apps so no need to double-check them. So I'm for removing this validation (Doorkeeper should just skip attributes which are not the part of the model schema)

JeremyC-za commented:
Ok, sounds good. I've removed the validation. Now, if we specify a custom attribute and either the grant/token models don't recognize it, then we'll get the exception `unknown attribute 'attribute_id' for Doorkeeper::AccessGrant`. As far as I'm concerned, that's fine, and should give developers enough information to fix. 

If we wanted to skip the attribute instead of letting the exception get raised, we could possibly do something like [this idea](https://github.com/doorkeeper-gem/doorkeeper/pull/1649/commits/34573fbf5cb1263eb100288a86a931d3477acb6b) I tried in an earlier commit, although I don't think it's necessary.

nbulaj reviewed (approved):
> unknown attribute 'attribute_id' for Doorkeeper::AccessGrant

That's totally fine for me :+1: Rails will do the validations instead of us :smile: 

Thanks!
